src/test/test-fileio.c

```diff
@@ -1034,7 +1034,10 @@ static void test_read_virtual_file(size_t max_size) {
                 size_t size = 0;
 
                 r = read_virtual_file(filename, max_size, &buf, &size);
-                log_info_errno(r, "read_virtual_file(\"%s\", %zu): %m (%zu bytes)", filename, max_size, size);
+                if (r < 0)
+                        log_info_errno(r, "read_virtual_file(\"%s\", %zu): %m", filename, max_size);
+                else
+                        log_info("read_virtual_file(\"%s\", %zu): %s (%zu bytes)", filename, max_size, r ? "non-truncated" : "truncated", size);
                 assert_se(r == 0 || ERRNO_IS_PRIVILEGE(r) || r == -ENOENT);
```

Comment:
This assertion should be also updated to support `r == 1`. Now, `r == 1` is handled accidentally in `ERRNO_IS_PRIVILGE()` macro.
```
if (r < 0) {
        log_info_errno(...);
        assert_se(ERRNO_IS_PRIVILEGE(r) || r == -ENOENT);
} else
        log_info(...);
```

Reply:
oh, indeed. hacking is hard!